### PR TITLE
Channel manager

### PR DESCRIPTION
**Pull request by LeftofZen**
A more well-defined mechanism between code asking to play audio, channels, and openal

### CODE REVIEW

File: src/OpenLoco/Audio/Audio.cpp [hunk 1]
```diff
@@ -818,21 +744,18 @@ namespace OpenLoco::Audio
...
     void stopVehicleNoise()
     {
-        for (auto& vc : _vehicleChannels)
-        {
-            vc.stop();
-        }
+        channelManager.stopChannels(ChannelId::vehicle);
     }
 
     void stopVehicleNoise(EntityId head)
```
Comment (LeftofZen) [inline src/OpenLoco/Audio/Audio.cpp:750]: move this to channel manager

File: src/OpenLoco/Audio/ChannelManager.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
+#include "ChannelManager.h"
+#include "VehicleChannel.h"
+#include <algorithm>
+#include <iterator>
+
+namespace OpenLoco::Audio
+{
+    static Channel* makeNewChannel(OpenAL::SourceManager& sourceManager, ChannelId channelId)
```
Comment (ZehMatt) [inline src/OpenLoco/Audio/ChannelManager.cpp:8]: Best to use unique_ptr, this is currently leaking.

File: src/OpenLoco/Audio/VehicleChannel.h [hunk 3]
```diff
@@ -4,16 +4,15 @@
 
 namespace OpenLoco::Audio
 {
-    class VehicleChannel
+    class VehicleChannel : public Channel
```
Comment (ZehMatt) [inline src/OpenLoco/Audio/VehicleChannel.h:7]: In my opinion its best to move out vehicle specific code out entirely, the audio code should be unaware of such things, the vehicle only updates the channel parameters so it's probably best to have that with the vehicle code and not here.

File: src/OpenLoco/Audio/ChannelManager.cpp (hunk 2 above)
Comment (LeftofZen) [reply src/OpenLoco/Audio/ChannelManager.cpp:9]: yeah I will, still trying to figure out why vehicle sounds aren't playing atm, then will fix the memory issues

File: src/OpenLoco/Audio/VehicleChannel.h (hunk 3 above)
Comment (LeftofZen) [reply src/OpenLoco/Audio/VehicleChannel.h:7]: agreed, I'll look into it

**Comment by AaronVanGeffen** [comment]
@LeftofZen This is now quite outdated. Would you like to continue this PR, or would you prefer to open another if you decide to continue?

**Comment by LeftofZen** [comment]
> @LeftofZen This is now quite outdated. Would you like to continue this PR, or would you prefer to open another if you decide to continue?

I'll close and resubmit when I get around to rewriting it

**Pull request closed without merging**